src/index.ts

```diff
@@ -1,36 +1,56 @@
-import {
-  WallyConnectorOptions,
-  RedirectOptions,
-} from './types';
+import { WallyConnectorOptions } from './types';
 
 import WallyConnector from './wally-connector';
 
-
 declare global {
   // eslint-disable-next-line no-var
-  var wally: WallyConnector
+  var wally: WallyConnector;
 }
 
+const checkInjected = () => {
+  if (!window.wally) {
+    console.error(
+      "Couldn't find wally instance. Ensure init() method is called first."
+    );
+    return false;
+  }
+  return true;
+};
+
 export const init = (options: WallyConnectorOptions): void => {
   if (typeof window === 'undefined') {
     console.error('Ensure init() is called on the client only.');
     return;
   }
   window.wally = window.wally || new WallyConnector(options);
-};
 
-export const handleRedirect = (options: RedirectOptions): void => {
-  if (!window.wally) {
-    console.error('Couldn\'t find wally instance. Ensure init() method is called first.');
-    return;
+  if (window.wally.isRedirected()) {
+    window.wally.handleRedirect();
   }
-  window.wally.handleRedirect(options);
-}
+
+  return;
+};
 
 export const getProvider = (): WallyConnector | null => {
-  if (!window.wally) {
-    console.error('Couldn\'t find wally instance. Ensure init() method is called first.');
+  if (!checkInjected()) {
     return null;
   }
+
   return window.wally;
-}
+};
+
+export const login = (): Promise<void> => {
```

Comment:
sure 👍 

I know you've got a few comments below about promises - any particular reason you aren't a fan? For simple stuff like this, I totally get it. But in some cases below I feel a little bit better explicitly using `resolve` and `reject` instead of an implicit `return`, which might not end up doing what you intend them to depending on the context